index.js

```diff
@@ -30,26 +42,23 @@ class BindingComponent extends React.Component {
         val = this.props.bindStateType(val);
       }
     }
-    const names = this.props.bindStateName.split('.');
-    if (names.length === 1) {
-      this.props.bindStateCtx.setState({
-        [names[0]]: val
-      });
-    } else {
-      const newState = this.props.bindStateCtx.state[names[0]];
-      const subnames = names.slice(1);
-      subnames.reduce((curr, name, index) => {
-        if (index === subnames.length - 1) {
-          curr[name] = val;
-        } else {
-          curr = curr[name];
-        }
-        return curr;
-      }, newState);
-      this.props.bindStateCtx.setState({
-        [names[0]]: newState
-      });
-    }
+
+    this.props.bindStateCtx.replaceState((prevState) => {
```

Comment:
As by definition, this _set bindStateVlaue_ should do be a deep-assign operation. This means that if we update a imaginary _price.value_ on a imaginary state:

```js
{
  "price": {
    "value": 20,
    "currency": "USD"
}
```

We _MUST_ leave _price.currency_ untouched.

Unfortunately the [official documentation](https://facebook.github.io/react/docs/component-api.html#setstate) didn't mention if _Component.setState_ is a deep-assign operation (or it wasn't until which version), for safety reason we have to deep-merge state manually.